lld/MachO/Arch/X86_64.cpp

```diff
@@ -96,11 +98,18 @@ int64_t X86_64::getEmbeddedAddend(MemoryBufferRef mb, uint64_t offset,
 void X86_64::relocateOne(uint8_t *loc, const Reloc &r, uint64_t value,
                          uint64_t relocVA) const {
   if (r.pcrel) {
-    uint64_t pc = relocVA + 4 + pcrelOffset(r.type);
+    uint64_t pc = relocVA + (1 << r.length) + pcrelOffset(r.type);
```

Comment:
Also, could we factor out `1 << r.length` into its own function? Then I could use it to fix the bug in `BPSectionOrderer.cpp` and hopefully prevent future bugs like it.

Reply:
a function just for one binary op?? seems like overkill haha

Reply:
It is also used in another file :) But yeah, it's not a hard requirement for me. Although I wish it was at least named `r.loglength` or something similar.